main/src/cgeo/geocaching/Trackable.java

```diff
@@ -48,6 +48,36 @@
     private TrackableConnector trackableConnector = null;
 
     /**
+     * Merge data from another Trackable.
+     * Squeeze existing data from the new one.
+     *
+     * @param newTrackable from which to pull informations
+     */
+    public void mergeTrackable(final Trackable newTrackable) {
+        guid = newTrackable.guid.equals("") ? guid : newTrackable.guid;
```

Comment:
StringUtils.defaultIfEmpty(...,...) can make this more readable by avoiding the repetition of the conditional.